packages/playout-gateway/src/atemUploader.ts

```diff
@@ -72,53 +88,67 @@ export class AtemUploadScript {
 			}
 		} else {
 			consoleLog('has no still')
-			throw Error('Atem appears to be missing still')
+			throw Error('Atem appears to be missing ' + type)
 		}
 	}
 
-	public async uploadToAtem(fileData: Buffer, stillIndex: number): Promise<void> {
-		const fileName = this.fileUrl.substr(-AtemMaxFilenameLength) // cannot be longer than 63 chars
-		if (!this.checkIfFileExistsOnAtem(fileName, stillIndex)) {
-			consoleLog('does not exist on ATEM')
+	public async uploadStillToAtem(fileName: string, fileData: Buffer, stillIndex: number): Promise<void> {
+		fileName = fileName.substr(-ATEM_MAX_FILENAME_LENGTH) // cannot be longer than 63 chars
+		if (!this.checkIfFileOrClipExistsOnAtem(fileName, stillIndex, AtemMediaPoolType.Still)) {
+			consoleLog(fileName + ' does not exist on ATEM')
 			await this.connection.clearMediaPoolStill(stillIndex)
 			await this.connection.uploadStill(stillIndex, fileData, fileName, '')
 		} else {
-			consoleLog('does exist on ATEM')
+			consoleLog(fileName + ' does exist on ATEM')
+		}
+	}
+	public async uploadClipToAtem(name: string, fileData: Buffer[], clipIndex: number): Promise<void> {
+		name = name.substr(-ATEM_MAX_CLIPNAME_LENGTH) // cannot be longer than 43 chars
+		if (!this.checkIfFileOrClipExistsOnAtem(name, clipIndex, AtemMediaPoolType.Clip)) {
+			consoleLog(name + ' does not exist on ATEM')
+			await this.connection.clearMediaPoolClip(clipIndex)
+			await this.connection.uploadClip(clipIndex, fileData, name)
+		} else {
+			consoleLog(name + ' does exist on ATEM')
```

Comment:
Shouldn't the else branch throw an exception to indicate that the clip has not been uploaded?